### PR TITLE
Add missing ark_mem frees if step mem malloc fails

### PR DESCRIPTION
**Pull request by Steven-Roberts**

### CODE REVIEW

**Comment by Steven-Roberts** [comment]
These missing frees were pretty benign since they only caused memory leaks for certain "create" function errors. To find these sneaky issues more systematically we might need a static analysis tool.

File: src/arkode/arkode.c [hunk 1]
```diff
@@ -133,6 +133,7 @@ ARKodeMem arkCreate(SUNContext sunctx)
   {
     arkProcessError(NULL, ARK_MEM_FAIL, __LINE__, __func__, __FILE__,
                     "Allocation of step adaptivity structure failed");
+    free(ark_mem);
```
Comment (gardner48) [inline src/arkode/arkode.c:136]:
```suggestion
    arkFree((void**)&ark_mem);
```

File: src/arkode/arkode.c [hunk 2]
```diff
@@ -144,6 +145,7 @@ ARKodeMem arkCreate(SUNContext sunctx)
   {
     arkProcessError(NULL, ARK_MEM_FAIL, __LINE__, __func__, __FILE__,
                     "Allocation of step controller object failed");
+    free(ark_mem);
```
Comment (gardner48) [inline src/arkode/arkode.c:148]:
```suggestion
    arkFree((void**)&ark_mem);
```

File: src/arkode/arkode.c [hunk 3]
```diff
@@ -180,6 +182,7 @@ ARKodeMem arkCreate(SUNContext sunctx)
   {
     arkProcessError(NULL, 0, __LINE__, __func__, __FILE__,
                     "Error setting default solver options");
+    free(ark_mem);
```
Comment (gardner48) [inline src/arkode/arkode.c:185]:
```suggestion
    arkFree((void**)&ark_mem);
```

**Review by gardner48 — CHANGES_REQUESTED** [review]

**Review by gardner48 — APPROVED** [review]

**Pull request merged**